### PR TITLE
New Bee: Chromatic Space Bee.

### PR DESCRIPTION
**Pull request by Nebulacrity**
## About the PR 
This PR adds this cute bee:
RGBee
It is created by injecting a bee egg with LSBee.

## Why's this needed? 
More bees is always good.

## Changelog

```
(u)Nebulacrity:
(*)A new colorful space bee type has been discovered recently!
```

### CODE REVIEW

**Review by pali6 — COMMENTED** [review]
That's a lot of duplicate code. I'd recommend maybe splitting the stuff you are overriding into separate procs on the base bee and then overriding just those bits.
E.g. make a proc `hug_back()` that adds `hugs` by default and then override it to also add `colors` etc.

**Comment by Nebulacrity** [comment]
Guess I'll start doing that soon then.

File: code/obj/critter/bee.dm [hunk 1]
```diff
@@ -1,3 +1,39 @@
+/obj/critter/domestic_bee/proc/do_hug(mob/user as mob)
+	user.visible_message("<span class='notice'>[src] hugs [user] back!</span>", "<span class='notice'>[src] hugs you back!</span>")
+	if (user.reagents)
+		user.reagents.add_reagent("hugs", 10)
+	return
+
+/obj/critter/domestic_bee/lsbee/do_hug(mob/user as mob)
```
Comment (ZeWaka) [inline code/obj/critter/bee.dm:7]: please put these procs next to their respective parents

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/critter/bee.dm (hunk 1 above)
Comment (ZeWaka) [reply code/obj/critter/bee.dm:7]: all the ones you added at the top that is

Comment (Nebulacrity) [reply code/obj/critter/bee.dm:7]: Hm, having problems doing so. Spawns a bunch of errors in a bunch of unedited files.

Comment (Nebulacrity) [reply code/obj/critter/bee.dm:7]: Ye, shit wack.

File: code/obj/critter/bee.dm [hunk 2]
```diff
@@ -1440,7 +598,794 @@
...
+		user.reagents.add_reagent("hugs", 3)
+		user.reagents.add_reagent("colors", 6)
+
+/obj/critter/domestic_bee/rgbee/do_reagentStuff(mob/M)
+	if (M.reagents.get_reagent_amount("methamphetamine") < 5)
+		M.reagents.add_reagent("methamphetamine", 1)
+	if (M.reagents.get_reagent_amount("colors") < 20)
+		M.reagents.add_reagent("colors", 5)
+
+/obj/critter/domestic_bee/lsbee
```
Comment (ZeWaka) [inline code/obj/critter/bee.dm:1382]: oof i accidentally deleted the lsbee reagentstuff and do_hug, can you fix?

Comment (Nebulacrity) [reply code/obj/critter/bee.dm:1382]: Uh yeah one sec.

Comment (Nebulacrity) [reply code/obj/critter/bee.dm:1382]: Done.

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**